sklearn/decomposition/tests/test_pca.py

```diff
@@ -529,6 +529,25 @@ def test_pca_score3():
     assert_true(ll.argmax() == 1)
 
 
+def test_pca_score4():
+    # Ensure that the scores are correctly calculated
+    # Specially designed for issue #7568, #8541, #8544
+    digits = datasets.load_digits()
+    X_digits = digits.data
+
+    pca1 = PCA(n_components=30, svd_solver='full', random_state=0)
```

Comment:
@lesteve Thanks. Corrected. According to the document, random_state is used when svd_solver == ‘arpack’ or ‘randomized’.